lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
@mikedodge04 I think what @btm is testing is running this in why-run. So in the example from your previous comment, the `cookbook_file "#{ld_path}#{daemon}.plist"` would not have have created the file, and then we'd get to `load_current_resource` and it'd blow up. As for why the assertions run after `load_current_resource`, I don't remember the reasoning for it. I'd guess that we have some assertions that rely on data from `load_current_resource` but I'd have to scan the code base to be sure.